src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/reverse_shape_and_type_infer.hpp"
+
+#include "itt.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace ov::opset10;
+
+namespace {
+bool inherit_output_shape(std::shared_ptr<ov::Node> node, std::vector<size_t> input_idxs) {
```

Comment:
Please avoid additional copies:
```suggestion
bool inherit_output_shape(const std::shared_ptr<ov::Node>& node, const std::vector<size_t>& input_idxs) {
```